sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ConnectionOptions.java

```diff
@@ -96,7 +127,16 @@ public ConnectionOptions(String fullyQualifiedNamespace, TokenCredential tokenCr
         CbsAuthorizationType authorizationType, AmqpTransportType transport, AmqpRetryOptions retryOptions,
         ProxyOptions proxyOptions, Scheduler scheduler, ClientOptions clientOptions,
         SslDomain.VerifyMode verifyMode, String product, String clientVersion, String hostname, int port) {
+        this(fullyQualifiedNamespace, tokenCredential,
```

Comment:
Why do we need these connection options?

Reply:
I'm not a fan of implicitly creating the TransactionCoordinator, as I see in your tests.  It defeats the purpose of exposing the interface method on AmqpConnection. 

```java
var coordinator = connection.getOrCreateTransactionCoordinator();
var transaction = connection.declareTransaction();

var session = coordinator.createSession("queue1");
var session2 = coordinator.createSession("queue2");

var producer = session.createProducer("queue1");
producer.send(message, transaction);

var producer2 = session2.createProducer("queue2");
producer2.send(message2, transaction);
```

Reply:
I would think of someone who would want to use azure-core-amqp stand-alone. as-is it is impossible to use without looking deeply at the code.

Reply:
connection options bag contains the property `distributedTransactionsSupport`  set by user at builder time , which is passed on to the Session object.  A session object need to know this property.
By using `ConnectionOption` bag to store this property, I was able to avoid additional API,  Your last comment was to use some option bag for this boolean property .

Reply:
I do not have this api `connection.getOrCreateTransactionCoordinator()`, This API is on AMQPSession. And I am now not creating implicit `Coordinator`, when an AMQP session is created. If you are talking about this test `ReactorConnectionTest`, I will remove this test.